docs/psoc6/quickref.rst

```diff
@@ -0,0 +1,670 @@
+    - From board manual? 
+    - From datasheet?
+    - To create a pinout diagram? 
+
+
+The driver is accessed via :ref:`machine.I2C <machine.I2C>`
+
+The constructor
+^^^^^^^^^^^^^^^
+An instance of the :mod:`machine.I2C` class can be created by invoking the constructor with all the 
+necessary parameters to fully configure the ``I2C``. By invoking the constructor, I2C peripheral is 
+initialized and configured to work in master mode. The maximum supported frequency is 1 MHz.
+
+::
+    
+    from machine import I2C
+    i2c = I2C(scl='P6_0', sda='P6_1', freq=400000)
+
+The ``scl`` and ``sda`` pins are the only mandatory arguments. The frequency is optional, and if not passed will be set to 400KHz by default.
+
+.. note::
+    The ``timeout`` option is currently not implemented in this port.
+
+::
+
+    from machine import I2C
+    i2c = I2C(scl='P6_0', sda='P6_1')  #I2C is initialized & configured with default frequency
+
+::
+
+    from machine import I2C
+    i2c = I2C(scl='P9_0', sda='P9_1', freq=400000)  #I2C is initialised & configured with given scl, sda pins & frequency
+
+Methods
+^^^^^^^
+All the methods(functions) given in :ref:`machine.I2C <machine.I2C>` class have been implemented in this port except:
+
+.. method:: I2C.init()
+
+All the initialization & configurations are handled by the constructor. Hence ``init()`` is not required.
+
+Hardware I2C bus slave
+----------------------
+
+The PSoC6™ port offers an additional class to implement an I2C slave device. An I2C master node connected to the slave can exchange data over I2C for the configured slave address and frequency.
+
+.. warning:: 
+    This is not part of the core MicroPython libraries. Therefore, not mapping any existing machine class API and neither supported by other ports.
+
+.. note:: 
+    | Part of the functionality of the I2C slave is based on hardware interrupts and callbacks. 
+    | As explained in this :ref:`section <isr_rules>`, writing interrupts handlers in MicroPython is subject to multiple considerations depending on the actual hardware capabilities and limitations, the specific port implementation, and the overall application design and implementation.
+    | These must kept in mind when implementing accurate timing and event synchronization between master and slave. 
+
+The constructor
+^^^^^^^^^^^^^^^
+
+.. class:: I2CSlave(scl, sda, addr, freq)
+
+    Constructs and returns a new I2C slave object using the following parameters.
+   
+    Required arguments: 
+      - *scl* should be a pin name supporting the SCL functionality.
+      - *sda* should be a pin name supporting the SDA functionality.
+      - *addr* should be an 8 bits unsigned integer. 
+
+    Optional arguments:
+      - *freq* should be an integer which sets the maximum frequency
+        for SCL. If not passed, by default is set to 400KHz.
+
+    Example:
+        ::
+            from machine import I2CSlave
```

Comment:
This is not rendered correctly as code block

Reply:
FIxed :)